### PR TITLE
I2C driver taken from PR #677, made functions static and aded burst modes

### PR DESCRIPTION
**Pull request by alignan**
Originally from PR #677, just uncristified, made some functions static and added burst modes for sending and transmitting.  Tested with the SHT25, TSL2563 and TMP102 sensors using the Re-Mote platform (CC2538 + CC1120 based).

### CODE REVIEW

**Comment by alignan** [comment]
@medmig @g-oikonomou

File: cpu/cc2538/dev/i2c.c [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+void
+i2c_init(uint8_t port_sda, uint8_t pin_sda, uint8_t port_scl, uint8_t pin_scl,
+         uint32_t bus_speed)
+{
+  volatile uint32_t ui32Delay;
+
+  /* Enable I2C clock in different modes */
+  REG(SYS_CTRL_RCGCI2C) |= 1; /* Run mode */
+  /* REG(SYS_CTRL_SCGCI2C) |= 1;	//Sleep mode */
+  /* REG(SYS_CTRL_DCGCI2C) |= 1;	//PM0 mode */
```
Comment (g-oikonomou) [inline cpu/cc2538/dev/i2c.c]: THose need removed

File: cpu/cc2538/dev/i2c.c [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+  /* Reset I2C peripheral */
+  REG(SYS_CTRL_SRI2C) |= 1; /* Reset position */
+
+  /* Delay for a little bit. */
+  for(ui32Delay = 0; ui32Delay < 16; ui32Delay++) {
+  }
```
Comment (g-oikonomou) [inline cpu/cc2538/dev/i2c.c]: I'd rather use a `clock_delay_usec` here

**Comment by alignan** [comment]
Changed!

**Comment by g-oikonomou** [comment]
:+1:

**Comment by medmig** [comment]
Thanks to @alignan for improving my old work, I'm glad that it has finally been merged :+1:

**Pull request merged**